src/main/java/seedu/address/model/contact/Contact.java

```diff
@@ -44,11 +44,7 @@ public boolean isSameContact(Contact otherContact) {
             return true;
         }
 
-        if (this.name == otherContact.name) {
-            return true;
-        }
-
-        if (otherContact.getName().equals(this.getName())) {
+        if (otherContact.getPhone().equals(this.getPhone())) {
```

Comment:
Would checking if both name and phone matches be good? I'm thinking of a scenario where in corporate there might be multiple contact sharing the same phone!

Reply:
I do see your point in the restriction. Will approve this PR and leave the merging to your astute judgement! 👍